### ISSUE TITLE
Comment animation mixed example code

### ISSUE THREAD

**Issue by Druidhawk** [mrdoob/three.js#7474]
given the importance, and the magnitude of the change, of the new animation mixer stuff.  I would appreciate some additional comments about what the various incantations are doing in each example.  they differ between the various examples.  I am sure they demonstrate various capabilities but I don't understand why each api was chosen in each example. one for example specifies an fops value, but only one of them, why? it's kind of like might first equalizer, lots of button that I don't know what they do and why you would use them.  appreciate the effort this took and want to maximize how I use it.

**Comment by mrdoob** [comment]
We're still working on it. The new API is "experimental", we'll document as soon as it gets more stable. I wouldn't want to document something that will change tomorrow.

Sorry about that.

**Comment by tschw** [comment]
> I don't understand why each api was chosen in each example

There have been various animation classes over time - and each example was written at some point. The then-current API has had an impact on how the examples were written and they have often just been ported over to the newer API. So there may not so much to understand, except that there are historical reasons.
Also, some examples in r73 still contain optimizations that are not advisable in general. If you see something that looks strange, see whether it's still present in the `dev` branch - if not, you're looking at a no longer needed optimization.

The `dev` / r74 version of the animation system has undergone a thorough overhaul, is several times faster and has a new (not yet frozen) API.

> first equalizer, lots of button that I don't know what they do and why you would use them

While the examples are nice to quickly get some code to play with, it's always best to look at the library classes to understand the design of the system. There is also quite a bit of documentation in the sources.

**Comment by Druidhawk** [comment]
Appreciate the responses :)  I will pull dev and play with that.  Once things settle down, I still think some guidance comments would be beneficial.

[**tschw**]I love reading API doc as much as anyone else :(.  In general, I think a simple comment along the lines of **//added animation in the form blah-blah because this is an X model format**", would help to understand the *intent* of the example's author.  Newbies tend to copy example code without understanding the rational behind it.  It may work, but they don't really know why.

In the three examples below, the author made API decisions, I would like to know why, FPS vs duration, why 'setLocalRoot'.

var clip = THREE.AnimationClip.CreateFromMorphTargetSequence( 'gallop', geometry.morphTargets, 30 );

mixer.addAction( new THREE.AnimationAction( geometry.animations[0] ).warpToDuration( duration ) );

	mixer.addAction( new THREE.AnimationAction( geometry.animations[0], Math.random() ).warpToDuration( duration ).setLocalRoot( mesh ) );

**Comment by princemio** [comment]
does the new API enable to add multiple skinnedmeshes to different scenes. I want to display an animated mesh AND render it to a texture at the same time. Is that possible?

**Comment by tschw** [comment]
@Druidhawk 

I renamed `.warpToDuration` to `.setDuration` - it adjusts the time scale so the clip takes a certain amount of time to play. 

The `.setLocalRoot` method has also been removed. If the clip is to be played on a different object than what the mixer is rooted on, you simply pass it as the second argument requesting the action:

```javascript
mixer.clipAction( clip, mesh ).setDuration( length ).play();
```

On `dev`, the example `webgl_animation_skinning_morph` has gotten a lot of GUI that lets you explore the new API and I added the new example `webgl_animation_authoring` just yesterday.

> does the new API enable to add multiple skinnedmeshes to different scenes. I want to display an animated mesh AND render it to a texture at the same time. Is that possible?

That might actually be another use case for `AnimationObjectGroup` (`dev` branch only, lets you apply the same animation state to different objects). Docs in the file header and  `webgl_shadowmap_performance` example currently uses it for optimization.
